Geometry/HcalCommonData/interface/HcalDDDRecConstants.h

```diff
@@ -80,29 +80,18 @@ class HcalDDDRecConstants {
        
 private:
   void                      initialize(void);
+  unsigned int              layerGroupSize( unsigned int eta ) const;
+  unsigned int              layerGroup( unsigned int eta, unsigned int i ) const;
 
-  bool                tobeInitialized;
-  static const int    nEtaMax=100;
   const HcalParameters      *hpar;
   const HcalDDDSimConstants &hcons;
-  int                 modeTopo_;  // Mode for topology
-  std::vector<int>    etaGroup;   // Eta Grouping
   std::vector<std::pair<int,int> > etaSimValu; // eta ranges at Sim stage
-  std::vector<double> etaTable;   // Eta table (HB+HE)
   std::vector<int>    ietaMap;    // Map Sim level ieta to Rec level ieta
-  std::vector<int>    iEtaMin, iEtaMax; // Minimum and maximum eta
-  std::vector<int>    maxDepth;   // Maximum depth in HB/HE/HF/HO           
-  int                 nEta;       // Number of bins in eta for HB and HE
-  std::vector<int>    phiGroup;   // Eta Grouping
   std::vector<double> phibin;     // Phi step for all eta bins (HB, HE, HO)
```

Comment:
@bsunanda - would it make sense to add the remaining vectors which are calculated at initialisation of this class to HcalParameters? The same for Sim part?